docs/blog/2019-08-14-strivectin-case-study/index.md

```diff
@@ -0,0 +1,134 @@
+
+- Clear separation between data definition and fetching
+- Easier visual testing
+
+## Shopify
+
+Shopify is an e-commerce platform with a robust Storefront API. The Content Management System (CMS) features in Shopify are limited without using its theme engine. The theme engine only works with the online store, meaning you can't use it for a headless Shopify store.
+
+We only needed to use Shopify for the transactional aspects of the site. This includes account management and checkout. The Storefront API supports checkout. We used the online store for account management.
+
+## Prismic
+
+Prismic is an API-first CMS. Users can define the shape of their content through "types". A powerful form builder allows users to describe the shape. The form fields can be rich text, options, grouped, repeatable, relations to other types, etc.
+
+Prismic also offers a "slices" feature. Different "types" can share the slices. This enables a Product and Homepage type to share a Hero Banner slice.
+
+The gatsby-source-prismic plugin integrates Gatsby with Prismic.
+
+## Firebase
+
+Google Cloud hosts our applications at Elevar. We wanted to stay in the same cloud provider, if possible. We settled on Cloud Build for CI/CD, but still needed a host. Firebase provided a solution to our problems. Firebase offers more than a static host. It offers HTTP functions for dynamic actions. This prevents you from having to think about another separate service.
+
+## Project plan and execution
+
+Our project plan was not traditional compared to most e-commerce projects:
+
+1. Design.
+2. Turn designs into components.
+3. Showcase components in Storybook with Knobs and Viewports.
+4. Use component data requirements (React Props) to create types in Prismic. Setup sample data for each type.
+5. Create components using data from components using Gatsby.
+6. Add integrations with Shopify and Third-parties.
+7. Configure Firebase and Launch site.
+
+After design approval, we could focus purely on the designs and component implementation. We were only in Storybook at this point. User Acceptance Testing (UAT) is simpler this way. The client can view components shared across pages in various states. Component tweaks are smoother during this phase. The component data is completely decoupled from the source. Our React components already defined the data that they needed. We only needed to define the data in the CMS (Prismic).
+
+After defining and populating our CMS with sample data, we were onto the Gatsby phase. Keep in mind that we didn't have any templates like a typical Shopify theme. You would think that it would be slower to start from scratch, but it had the opposite effect. Implementing with Gatsby enabled us to developer faster. We weren't bound by templates like a typical Shopify theme. We were able to reuse queries and components across pages.
```

Comment:
```suggestion
After defining and populating our CMS with sample data, we were onto the Gatsby phase. Keep in mind that we didn't have any templates like a typical Shopify theme. You would think that it would be slower to start from scratch, but it had the opposite effect. Implementing with Gatsby enabled us to develop faster. We weren't bound by templates like a typical Shopify theme. We were able to reuse queries and components across pages.
```

Reply:
Updated in 046c737